sql-plugin/src/main/scala/com/nvidia/spark/rapids/SparkShims.scala

```diff
@@ -78,6 +78,7 @@ trait SparkShims {
       readFunction: (PartitionedFile) => Iterator[InternalRow],
       filePartitions: Seq[FilePartition],
       readDataSchema: StructType,
+      relation: Option[HadoopFsRelation],
```

Comment:
if you move this arg to the last position and give it a default value None, you probably will have fewer lines to modify